software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java

```diff
@@ -134,17 +189,20 @@ protected void createDirectory(String directoryName, String summaryForLogging) {
         getLocation().executePsScript("New-Item -path \"" + directoryName + "\" -type directory -ErrorAction SilentlyContinue");
     }
 
-    protected WinRmToolResponse executeCommand(ConfigKey<String> regularCommandKey, ConfigKey<String> powershellCommandKey, boolean allowNoOp) {
-        String regularCommand = getEntity().getConfig(regularCommandKey);
-        String powershellCommand = getEntity().getConfig(powershellCommandKey);
-        if (Strings.isBlank(regularCommand) && Strings.isBlank(powershellCommand)) {
+    public int executeCommandInTask(String regularCommand, String powershellCommand, String phase, Boolean allowNoOp) {
```

Comment:
It is surprising that the presence of the `Boolean allowNoOp` in the overloaded method causes it to go down a completely different code path, not calling `newScript(...)...` in the same was as the other `executeCommandInTask` method.

Do you really mean for this `executeCommandInTask` method that it is *not* executed in a task? Or should it do the no-op guard, and then delegate to the other `executeCommandInTask` method?